gwcs/geometry.py

```diff
@@ -55,3 +56,91 @@ def evaluate(self, cosa, cosb, cosc, length):
 
     def inverse(self):
         return ToDirectionCosines()
+
+
+class SphericalToCartesian(Model):
+    """
+    Convert spherical coordinates on a unit sphere to cartesian coordinates.
+    Spherical coordinates, when not provided as ``Quantity``, are assumed
+    to be in degrees with ``phi`` being the *azimuthal* angle ``[0, 360)``
+    and ``theta`` being the *elevation* angle ``[-90, 90]``.
+
+
+
+    """
+    _separable = False
+
+    _input_units_strict = True
+    _input_units_allow_dimensionless = True
+
+    n_inputs = 2
+    n_outputs = 3
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.inputs = ('phi', 'theta')
+        self.outputs = ('x', 'y', 'z')
+
+    @staticmethod
+    def evaluate(phi, theta):
+        if isinstance(phi, u.Quantity) + isinstance(theta, u.Quantity) == 1:
+            raise TypeError("All arguments must be of the same type "
+                            "(i.e., quantity or not).")
+
+        phi = np.deg2rad(phi)
+        theta = np.deg2rad(theta)
+
+        cs = np.cos(theta)
+        x = cs * np.cos(phi)
+        y = cs * np.sin(phi)
+        z = np.sin(theta)
+
+        return x, y, z
+
+    def inverse(self):
+        return CartesianToSpherical()
+
+    @property
+    def input_units(self):
+        return {'phi': u.deg, 'theta': u.deg}
+
+
+class CartesianToSpherical(Model):
+    """
+    Convert cartesian coordinates to spherical coordinates on a unit sphere.
+    Spherical coordinates are assumed to be in degrees with ``phi`` being
+    the *azimuthal* angle ``[0, 360)`` and ``theta`` being the *elevation*
+    angle ``[-90, 90]``.
+
+    """
+    _separable = False
+
+    _input_units_strict = True
+    _input_units_allow_dimensionless = True
```

Comment:
Since this class does not define `input_units` there's no need to set the above two attributes. Sorry for misleading you.